octavia_f5/restclient/as3classes.py

```diff
@@ -305,3 +305,11 @@ class HTTP_Profile(BaseDescription):
     def __init__(self, **kwargs):
         super(HTTP_Profile, self).__init__(locals())
         setattr(self, 'class', 'HTTP_Profile')
+
+
+class Data_Group(BaseDescription):
+    def __init__(self, _records, keyDataType='ip', **kwargs):
+        super(Data_Group, self).__init__(locals())
```

Comment:
Any particular reason you're using `locals()` here? I've never seen this style before and don't think it's necessary.

Reply:
This is not my idea. We do it in this file everywhere
<img width="836" alt="image" src="https://github.com/sapcc/octavia-f5-provider-driver/assets/8326634/d80d8ee1-079a-4e8f-97a3-cc75833e2026">

And we do it because the parent class `BaseDescription` will unpack these variables and set them as attributes of object https://github.com/sapcc/octavia-f5-provider-driver/blob/stable/yoga-m3/octavia_f5/restclient/as3classes.py#L36

Reply:
Cool, thanks